n2v/methods/zmp.py

```diff
@@ -248,22 +252,34 @@ def zmp_scf(self,
                 if SCF_ITER == maxiter - 1:
                     raise ValueError("Maximum Number of SCF cycles reached. Try different settings.")
 
-            density_current = self.on_grid_density(grid=None, Da=Da, Db=Db, vpot=self.vpot)
-            grid_diff = np.max(np.abs(D0 - density_current))
-            if np.abs(grid_diff_old) < np.abs(grid_diff):
-                print("\nZMP halted. Density Difference is unable to be reduced")
-                break
-
-            grid_diff_old = grid_diff
-            print(f"SCF Converged for lambda:{int(lam_i):5d}. Max density difference: {grid_diff}")
-
-            self.Da = Da
-            self.Db = Db
+            # Why do we need this? Basis set -> grid is expensive.
```

Comment:
I know there is a relationship between the accuracy of P_nm and n(r), but they're not identical. I'd still find value in having the accuracy on the grid. Maybe just outside scf?

Reply:
Agree. But I am not sure if we want to break when the density error stops updating.